mkosi/__init__.py

```diff
@@ -863,7 +864,8 @@ def sign_efi_binary(context: Context, input: Path, output: Path) -> Path:
         context.config.secure_boot_sign_tool == SecureBootSignTool.auto and
         find_binary("sbsign", root=context.config.tools()) is not None
     ):
-        with open(output, "wb") as f:
+        with tempfile.NamedTemporaryFile(dir=output.parent, prefix=output.name) as f:
```

Comment:
I think this can raise `FileNotFoundError`, since the file is missing when it's supposed to be deleted, at least this happens with
```python
with tempfile.NamedTemporaryFile(prefix="foo-", delete=False) as f:
     f.write(b"foobar\n")
     os.rename(f.name, "/tmp/foobar")
```
for me. `NamedTemporaryFile` has a `delete=` argument, that is `True` by default, setting it to false should make this safe.

Reply:
Setting it to `false` means the file doesn't get cleaned up on failure either. Changed it to use `os.link()` instead.